ports/stm32/boards/Passport/modules/pages/pin_entry_page.py

```diff
@@ -258,12 +260,20 @@ def on_key(self, event):
             self.show_brick_warning()
 
     async def on_security_words(self, security_words, error):
+        from serializations import sha256
         # NOTE: Be aware that this is called from the context of another task
         if error is None:
             self.security_words = security_words
             self.show_security_words = True
             if self.show_security_words and self.user_wants_to_see_security_words:
-                self.update_message(show_security_words=True, title=self.security_words_message)
+                new_pin_sha = sha256(self.pin)
+                true_pin_sha = common.settings.get('first_four_hash')
+                if self.check_first_four and not all(x == y for x, y in zip(new_pin_sha, true_pin_sha)):
+                    self.security_words_message = ("Your first four digits are wrong.\n"
```

Comment:
Maybe change to "Your pin prefix is wrong."  Consider if we want a different user-facing name for these digits.

Reply:
Are the users ever introduced to the idea of the first four digits being called "prefix"? If not, it could be confusing to refer to a prefix they haven't heard of.

Reply:
No, which is why I suggested we consider what the user-facing name should be.  Discuss with @zachherbert.

Reply:
Maybe we just say that the PIN is incorrect instead.